### ISSUE TITLE
False positive E1126 when changing list to NumPy array inside dict

### ISSUE THREAD

**Issue by otaithleigh** [PyCQA/pylint#2569]
Pylint raises error E1126 after changing a `list` to an `ndarray` inside a `dict`. The issue appears at least partly to be the multiple indices; 1-dimensional indices are fine.

### Steps to reproduce
1. Script:
```python
import numpy as np
d = {'k': [[1, 2, 3]]}
d['k'] = np.array(d['k'])
d['k'][:, 0]  # <== Raises error E1126
```
2. pylint output:
```
$ pylint -E test_pylint_error.py
************* Module test_pylint_error
test_pylint_error.py:4:0: E1126: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)
```
3. Interpreter output:
```python
>>> import numpy as np
>>> d = {'k': [[1, 2, 3]]}
>>> d['k'] = np.array(d['k'])
>>> d['k'][:, 0]
array([1])
```

### Current behavior
Raises error `E1126`.

### Expected behavior
No error; code runs fine in interpreter.

### pylint --version output
```
pylint 2.1.1
astroid 2.0.2
Python 3.7.0 (default, Aug  1 2018, 03:10:49)
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
Thanks for reporting an issue. Unfortunately pylint cannot know you are reassigning the type of `k` with `d['k'] = np.array(d['k'])` which is why it is confused. A solution would be to set that type right from the start, but you can also disable the error you have in place.

**Comment by otaithleigh** [comment]
Ah, ok. Thanks!
